model/src/main/proto/survey.proto

```diff
@@ -0,0 +1,182 @@
+enum QuestionName{
+  // The question name is unknown.
+   QUESTION_NAME_UNSPECIFIED = 0;
+  // Corresponds to the name of the user type question.
+  QUESTION_USER_TYPE = 1;
+
+  //  Corresponds to the market fit question.
+  QUESTION_MARKET_FIT = 2;
+
+  // Corresponds to  the NPS question.
+  QUESTION_NPS = 3;
+
+  // Corresponds to  the open-ended question for a NPS promoter.
+  QUESTION_PROMOTER_FEEDBACK = 4;
+
+  // Corresponds to  the open-ended question for a NPS passive.
+  QUESTION_PASSIVE_FEEDBACK = 5;
+
+  // Corresponds to  the open-ended question for a NPS detractor.
+  QUESTION_DETRACTOR_FEEDBACK = 6;
+}
+
+// Represents the multiple choice answer available for a question.
+message SurveyQuestionAnswer{
+  // The ID of this answer option.
+  string question_answer_id = 1;
+
+  // The semantic name of the question this answer option is tied to.
+  QuestionName question_name = 2;
+
+  // The actual text value of the answer.
+  string answer = 3;
+}
+
+// Represents an answer that was selected by a user for a survey question.
+message SurveyUserAnswer{
+  //  The ID of the survey to which this answer is tied.
+  string  survey_id = 1;
+
+  // Semantic name of the question the response is tied to.
+  QuestionName question_name = 2;
+
+  // The value of the answer the user gave.
+  string user_answer = 3;
+}
+
+// Represents the different types of users who could be taking the survey.
+// Consists of answer options for the user type question.
+enum  UserTypeAnswer{
+  // User type unknown.
+  USER_TYPE_UNSPECIFIED =0;
+  // Corresponds to an unspecified or unknown type of user.
+  OTHER = 1;
+
+  // Corresponds to a user who is a teacher.
+  TEACHER = 2;
+
+  // Corresponds to a user who is a parent.
+  PARENT = 3;
+
+  // Corresponds to a user who is a learner.
+  LEARNER = 4;
+}
+
+// Represents general opinion of the user about the Oppia android app.
+// Consists of answer options for the market fit question.
+enum  MarketFitAnswer{
+  // Answer unknown.
+  MARKET_FIT_ANSWER_UNSPECIFIED =0;
+
+  // Corresponds to the answer for Very Disappointed.
+  VERY_DISAPPOINTED = 1;
+
+  // Corresponds to the answer for Disappointed.
+  DISAPPOINTED = 2;
+
+  // Corresponds to the answer for  Not Disappointed.
+  NOT_DISAPPONTED = 3;
+
+  // Corresponds to the answer for  N/A - Don't use Oppia Anymore.
+  NA_DONT_USE_OPPIA_ANYMORE = 4;
+}
+
+// Represents the status of the survey at the time it is exited.
+enum SurveyCompletionStatus{
+  // The completion status is unknown.
+  SURVEY_COMPLETION_STATUS_UNSPECIFIED = 0;
+
+  // Corresponds to the survey being declined, not started.
+  SURVEY_DECLINED = 1;
+
+  // Corresponds to the survey being exited before the mandatory questions are completed.
+  SURVEY_STARTED_NOT_COMPLETED = 2;
+
+  // Corresponds to the mandatory part of the survey being completed.
+  SURVEY_MANDATORY_PART_COMPLETED = 3;
+
+  // Corresponds to the optional part of the survey being completed.
+  SURVEY_COMPLETED = 4;
```

Comment:
Perhaps 'FULLY_COMPLETED' might better correspond to 'MANDATORY_PART_COMPLETED'?

Reply:
This sounds good.